File: Data Structures & Algorithms/remove-element/submission-0.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int unmatchedCursor = 0;
        for(int i = 0; i < nums.length; ++i) {
            if(nums[i] != val) {
                nums[unmatchedCursor] = nums[i];
                unmatchedCursor++;
            }
        }

        return unmatchedCursor;
    }
}

File: Data Structures & Algorithms/sort-colors/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {void} Do not return anything, modify nums in-place instead.
     */
    sortColors(nums) {
        const buckets = []

        for (const num of nums) {
            buckets[num] ??= 0
            buckets[num] = buckets[num] + 1
        }


        let idx = 0
        for (let i = 0; i < buckets.length; i++) {
            for (let j = 0; j < buckets[i]; j++) {
                nums[idx] = i
                idx++
            }
        }

    }
}
